site/content/api/Device-Connection-API.md

```diff
@@ -0,0 +1,148 @@
+
+| Name         | Mandatory | Location                 | AMQP Type | Description |
+| :----------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*    | yes       | *properties*             | *string*  | MUST be set to `setLastKnownGatewayForDevice`. |
+| *gateway_id* | yes       | *application-properties* | *string*  | The identifier of the gateway that last acted on behalf of a given device. If a device connects directly instead of through a gateway, the device identifier itself is to be used here. |
+
+The body of the message SHOULD be empty and will be ignored if it is not.
+
+**Response Message Format**
+
+A response to a *set last known gateway for device* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message's *status* property may contain the following codes:
+
+| Code | Description |
+| :--- | :---------- |
+| *204* | OK, the last known gateway for the device was updated. |
+| *404* | Not Found, there is no device registered with the given *device_id* within the given *tenant_id*. |
+
+For status codes indicating an error (codes in the `400 - 499` range) the message body MAY contain a detailed description of the error that occurred.
+
+## Get Last Known Gateway for Device
+
+Clients use this command to *retrieve* the gateway that last acted on behalf of a given device.
+
+This operation is *mandatory* to implement.
+
+**Message Flow**
+
+
+The following sequence diagram illustrates the flow of messages involved in a *Client* retrieving the last known gateway for a Device.
+
+![Get Last Known Gateway for Device message flow](../getLastKnownGatewayForDevice_Success.png)
+
+
+**Request Message Format**
+
+The following table provides an overview of the properties a client needs to set on a message to retrieve the last known gateway for a device in addition to the [Standard Request Properties]({{< relref "#standard-request-properties" >}}).
+
+| Name        | Mandatory | Location                 | AMQP Type | Description |
+| :---------- | :-------: | :----------------------- | :-------- | :---------- |
+| *subject*   | yes       | *properties*             | *string*  | MUST be set to `getLastUsedGateway`. |
+
+The body of the message SHOULD be empty and will be ignored if it is not.
+
+**Response Message Format**
+
+A response to a *get last known gateway for device* request contains the [Standard Response Properties]({{< relref "#standard-response-properties" >}}).
+
+The response message includes payload with a JSON object having the following properties:
+
+| Name             | Mandatory | JSON Type     | Description |
+| :--------------- | :-------: | :------------ | :---------- |
+| *device-id*      | *yes*     | *string*      | The ID of the last known gateway for the device, or the device ID itself if the device doesn't support usage via a gateway. |
```

Comment:
then this should probably be named *gateway-id*, right?

Reply:
I first thought that because the returned id could also be the device ID (if the device isn't connected to a gateway) the more general `device-id` should better be used.
But I agree that this could be confusing, also since in the setter operation the property is also named `gateway_id`.
I'll change it to `gateway-id`.